src/routes/safe/components/Apps/components/ConfirmTransactionModal.tsx

```diff
@@ -118,6 +121,10 @@ const ConfirmTransactionModal = ({
 
   const areTxsMalformed = txs.some((t) => !isTxValid(t))
 
+  if (!ethToken) {
```

Comment:
Why do we not return already in the same place where we check `isOpen`? Right now we are still setting up all the functions before we realize that we should not even render the modal.

Also should we use react hooks for all the functions we create here?